core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -777,9 +777,11 @@ private static Expression implementNullSemantics(
       // v0 != null && v1 != null && f(v0, v1)
       for (Ord<RexNode> operand : Ord.zip(call.getOperands())) {
         if (translator.isNullable(operand.e)) {
-          list.add(
-              translator.translate(
-                  operand.e, NullAs.IS_NOT_NULL));
+          Expression te = translator.translate(operand.e, NullAs.IS_NOT_NULL);
+          Class tec = (Class) te.getType();
+          if (!tec.isPrimitive() || tec.equals(boolean.class)) {
```

Comment:
@vlsi I moved the comments into JIRA.  the issue is manifested as `byte` object is  emitted into the    `not null` expression. but `byte` primitive object itself is already `not null`. so primitives should be skipped. 

anyways Julian mentioned this is not worth fixing using this approach at the moment.